daml/Marketplace/Issuance/Service.daml

```diff
@@ -57,13 +55,14 @@ template Service
         with
           createOriginationCid: ContractId OriginationRequest
         do
-          OriginationRequest{assetLabel; cfi; description; claims; safekeepingAccount; observers} <- fetch createOriginationCid
+          OriginationRequest{assetLabel; cfi; description; claims; observers} <- fetch createOriginationCid
           -- assertMsg "issued must be 0" $ description.issued == 0.0
           -- assertMsg "authorized must be > 0" $ description.authorized > 0.0
           archive createOriginationCid
-          let
-            assetId = Id with signatories = fromList [ provider, customer ]; label = assetLabel; version = 0
-            issuer = customer
+          let assetId = Id with signatories = fromList [ provider, customer ]; label = assetLabel; version = 0
+              issuer = customer
+              registrar = provider
+
```

Comment:
Note: implicit assumption here that the provider of the issuance service is the same as the registrar (root custodian) of the asset at hand. I think it's reasonable to assume that but we should think about possible situations where this might not be the case (and if / how it can be supported)

Reply:
This is a very important point - because we also make this assumption in multiple places in the UI when determining the correct `Custody.Service` to utilise or when determining the correct delivery account. We assume the `Issuance.Service` provider is the `registrar` on the `AssetDescription` and inherently the `Custody.Service` provider are the same party.